core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -159,18 +169,124 @@ public void commit(TableMetadata base, TableMetadata metadata) {
     int nextVersion = (current.first() != null ? current.first() : 0) + 1;
     Path finalMetadataFile = metadataFilePath(nextVersion, codec);
     FileSystem fs = getFileSystem(tempMetadataFile, conf);
+    boolean versionCommitSuccess = false;
+    boolean useObjectStore =
+        metadata.propertyAsBoolean(
+            TableProperties.OBJECT_STORE_ENABLED, TableProperties.OBJECT_STORE_ENABLED_DEFAULT);
+    int previousVersionsMax =
+        metadata.propertyAsInt(
+            TableProperties.METADATA_PREVIOUS_VERSIONS_MAX,
+            TableProperties.METADATA_PREVIOUS_VERSIONS_MAX_DEFAULT);
+    // todo:Currently, if the user is using an object store, we assume that he must be using the
+    //  global locking service. But we should support add some other conditions in future.
+    boolean supportGlobalLocking = useObjectStore;
+    try {
+      tryLock(tempMetadataFile, metadataRoot());
+      versionCommitSuccess =
+          commitNewVersion(
```

Comment:
Executing the commitNewVersion method can only produce four results:
1.return true.(rename success)
2.return false.(rename failed)
3.throw CommitFailedException
4.throw CommitStateUnknownException